modules/aggregation-layers/src/contour-layer/marching-squares.js

```diff
@@ -56,6 +60,7 @@ export function getCode(opts) {
   // TOP
   if (isLeftBoundary || isTopBoundary) {
     codes.top = 0;
+    weights.top = 0;
```

Comment:
I think this may be a problem, when processing the boundary, we don't want to perform interpolation , we want the intersection point to be on the boundary, please investigate, you can use the demo to verify, with and without interpolation.